content/2016.12.how-to-develop-a-webapp-with-Kinto.md

```diff
@@ -0,0 +1,247 @@
+        "author": "Jon Snow",
+        "id": "a1e6e879-5c94-4a01-9ec2-28e152d1d8f4"
+    },
+    {
+        "message": "I know that I know nothing",
+        "last_modified": 1480340396521,
+        "author": "Socrates",
+        "id": "94787c48-aa96-4121-879f-1bd68d3c1f23"
+    }
+]
+
+getMessage(message, author) {
+    return (
+    `<li>
+      <span>
+        <span>${author}</span>
+        <span>${message}</span>
+      </span>
+    </li>`
+    );
+}
+
+displayMessage({message, author}) {
+    var container = document.createElement('div');
+    container.innerHTML = this.getMessage(message, author);
+    this.messagesList.firstChild.appendChild(container.firstChild);
+}
+```
+
+Here is how the HTML code looks like:
+
+```html
+<div id="messages">
+  <ul>
+    <li>
+      <span>
+        <span>Jon Snow</span>
+        <span>I know nothing</span>
+      </span>
+    </li>
+    <li>
+      <span>
+        <span>Socrates</span>
+        <span>I know that I know nothing</span>
+      </span>
+    </li>
+  </ul>
+</div>
+```
+
+## Post a new message
+
+![](images/kintochat-new-post.png)
+{.image}
+
+This time, we only have to send the form's data to the Kinto server. Once again, we could go with a simple POST request `https://kinto.discute.re/v1/buckets/chat/collections/4815162342/records` but here we will use the Kinto API to write:
+
+```js
+const messageInput = document.getElementById('message');
+const messageForm = document.getElementById('message-form');
+messageForm.addEventListener('submit', saveMessage);
+
+saveMessage(e) {
+    e.preventDefault();
+    if(messageInput.value) {
+        client.bucket('chat')
+            .collection('4815162342')
+            .createRecord({
+                author: 'Jean Neige',
+                message: messageInput.value
+            });
+    }
+}
+```
+
+## Front-end enhancements
+
+Finally to add some nice enhancements, we use:
+
+- [getmdl.io](http://getmdl.io) to syle the components;
+- [pusher.com](http://pusher.com) to automatically update our app when new messages arrive.
+
+With these modifications and without having to write any backend code, we have a WebApp accessible at this URL: [https://enguerran.github.io/kintochat/#4815162342](https://enguerran.github.io/kintochat/#4815162342).
+
+The example [source code is available on Github](https://github.com/enguerran/kintochat), please keep in mind that it is just for educational purpose, it’s not production ready, it’s not even covered with tests.
+
+## But, what is Kinto?
+
+You can think of Kinto as a backend API service to store and sync your data.
+
+Kinto is written in Python around [Pyramid](https://trypyramid.com/). Kinto is highly configurable through its `kinto.ini` file and [a set of plugins](https://github.com/Kinto/kinto/wiki/Plugins).
+
+Kinto is [OpenSource](https://github.com/kinto) and allows us to store of our user's data away from Google, Amazon, Apple or Microsoft closed silos and other private cloud unicorns.
```

Comment:
The thing is, they do a lot of different things, so maybe we should be more specific and target some of their products; google docs, apple icloud, amazon rds and so on